src/kademlia/find_data.cpp

```diff
@@ -135,8 +135,6 @@ char const* find_data::name() const { return "find_data"; }
 
 void find_data::done()
 {
-	if (m_invoke_count != 0) return;
```

Comment:
No, `m_invoke_count may` be non-zero here. Based on the fact that done() is called based on the return value from `traversal_algorithm::add_requests()`, I believe the intent was for done() to be called only when the entire traversal is done, not just a single lookup.

It looks like this line was added as a workaround for the fact that done() will be called multiple times if add_requests() returns true but `m_invoke_count` is still non-zero.

I think it's better to deal with this inside `traversal_algorithm` rather than force every child class to deal with done() being called multiple times. Moreover, it is highly likely that at least one of the outstanding queries when add_requests() declares the traversal done will timeout. This means the actual done() processing gets delayed until *all* queries receive a response or timeout. I believe this was why @thomasyuan was seeing *every* mutable get request take 15 seconds to complete. It is better to avoid this delay by executing all the done() processing as soon as the K closest nodes have responded, then manipulate the flags of the remaining queries so that they do not cause spurious calls to done() when they timeout or receive a response.